### ISSUE TITLE
Lack of specification for ERROR messages

### ISSUE THREAD

**Issue by angiolep** [wamp-proto/wamp-proto#256]
The following specification sections:

- Subscribe ERROR
- Unsubscribe ERROR
- Publish ERROR
- Register ERROR
- Invocation ERROR
- Call ERROR

are all about the ERROR messages a router shall reply to their clients when the relevant requests cannot be fulfilled. They all define the __format__ of the ERROR messages (specifying their fields with their types and semantic)  and give on relevant example.

Unfortunately, __none of them clearly specify under which circumstances__ (for which specific scenarios) the router MUST NOT (or SHOULD NOT as per [RFC2119](https://tools.ietf.org/html/rfc2119)) fulfill the request. 

Because of this lack of specification, the readers are left

- to guess the reasons why a router shall not fulfill a request and reply ERROR,
- to guess which error URI the router shall reply amongst those few specified,
- to design custom error URIs when none of those few specified seem to be the right one.

For example, I assumed that a router shall not fulfill a REGISTER request if the given procedure URI has been already registered and I decided to reply ``wamp.error.procedure_already_exists`` because 

- that's the only example given in section "Register ERROR" and ``procedure_already_exists`` made me guess it could mean _"already registered"_
- I found it _"mentioned"_ in section "Predefined URIs / Basic Profile / Interactions"

I personally would have preferred to read clear statements in section "Register ERROR" that would have made me 100% sure that's the behaviour to implement for the Basic Profile as I guess it could not be right behaviour for the Advanced Profile (which provides _"shared registrations"_ instead)

**Comment by oberstet** [comment]
I agree, the spec text is clearly lacking in this respect. In multiple ways.
